synapse/lib/config.py

```diff
@@ -70,8 +72,15 @@ def getJsValidator(schema):
         return func
 
     func = fastjsonschema.compile(schema)
-    _JsValidators[key] = func
-    return func
+
+    def wrap(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except JsonSchemaException as e:
+            raise s_exc.BadJson(mesg=e.message, name=e.name)
```

Comment:
i feel like there is a slight diff between "BadJson" and "Your schema was invalid"

Reply:
the fastjson library raises JsonSchemaException when the thing being validated ( not the schema itself ) is invalid